wiki/wiki/doctype/wiki_page/wiki_page.py

```diff
@@ -216,6 +216,23 @@ def calculate_toc_html(self, html):
 	def get_context(self, context):
 		self.verify_permission("read")
 		self.set_breadcrumbs(context)
+		ccc = frappe.db.sql(""" select user, allow, for_value,name from `tabUser Permission` where user=%s and allow='Wiki Space' """,(frappe.session.user))
```

Comment:
```suggestion
		ccc = frappe.db.get_value("User Permission", {"user": frappe.session.user, "allow": "Wiki Space"}, ["user", "allow", "for_value", "name"])
```

Try to use [database api](https://frappeframework.com/docs/user/en/api/database) and [query builder](https://frappeframework.com/docs/user/en/api/query-builder) as much as possible instead of raw sql 
Also use better variable names. `ccc` is very ambigous

Reply:
you have to rewrite it everywhere you're using `frappe.db.sql`